embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/delivery/EmbraceCacheService.kt

```diff
@@ -149,6 +148,24 @@ internal class EmbraceCacheService(
         return null
     }
 
+    /**
+     * Gets a file from the files directory.
+     * Generally used to write files, so we store them in the files directory.
+     * The files directory is the default directory.
+     */
+    private fun getFileFromFilesDir(name: String): File {
```

Comment:
Do you think we can hide the logic of where the files are stored within the `StorageService`? If that part is abstracted away fully, `EmbraceCacheService` should haven't to know where the files are stored. Is there a reason it needs to? In other words, can we defer all the "where the file is stores" logic to the storage service and have this completely not know about it?

Reply:
As we discussed, I ended up dividing the implementation of getFile into two methods:
- getFileForRead
- getFileForWrite
This way we hide completely the implementation details related to the storage directories.